pkg/mesh/topology.go

```diff
@@ -128,14 +133,26 @@ func NewTopology(nodes map[string]*Node, peers map[string]*Peer, granularity Gra
 			// - the node's allocated subnet
 			// - the node's WireGuard IP
 			// - the node's internal IP
+			// - IPs that were specified by the allowed-location-ips annotation
 			allowedIPs = append(allowedIPs, node.Subnet)
+			for _, ip := range node.AllowedLocationIPs {
+				allowedLocationIPsMap[ip.String()] = ip
+			}
 			if node.InternalIP != nil {
 				allowedIPs = append(allowedIPs, oneAddressCIDR(node.InternalIP.IP))
 				privateIPs = append(privateIPs, node.InternalIP.IP)
 			}
 			cidrs = append(cidrs, node.Subnet)
 			hostnames = append(hostnames, node.Name)
 		}
+		var allowedLocationIPs []*net.IPNet
+		for _, ip := range allowedLocationIPsMap {
+			allowedLocationIPs = append(allowedLocationIPs, ip)
```

Comment:
instead of iterating another time here, how about we use the map only as a set for membership checks `map[string]{}` and append to the slice if the allowed IP is not yet in the set?